### PR TITLE
Enhance the option lines in output

### LINKED ISSUE #382: Add parameter `--no-trusted-host`: Don't add trusted host to generated file
Describe the issue briefly here.
##### Steps to replicate
1. Prepare `~/.pip/pip.conf`

``` ini
[global]
index-url = https://pypi.tuna.tsinghua.edu.cn/simple
extra-index-url = http://devpi.arpanet/private/dev/+simple
trusted-host = devpi.arpanet

[search]
index = http://devpi.arpanet/private/dev/
```

Then
2. `pip-compile --no-index --upgrade --output-file requirements.txt requirements.in`
##### Expected result

`--trusted-host devpi.arpanet` should not existed in `requirements.txt`
##### Actual result

I wish there's a option in `pip-sync` like `--no-index` which won't write `--trusted-host devpi.arpanet` into the generated `requirements.txt`.

### PR DESCRIPTION
**Pull request by tonyseek**
### Description

There is a proposal to fix following output:

```
--index-url http://pypi.example.com/root/pypi
--extra-index-url http://pypi.example.com/root/pypi
--extra-index-url http://pypi.example.com/root/pypi
--trusted-host pypi.example.com
--trusted-host pypi.example.com
```

Those output may be caused by multiple option sources (e.g. `env PIP_INDEX_URL=$U PIP_EXTRA_INDEX_URL=$U pip-compile --extra-index-url=$U x.in`).

### Changes

- Deduplicate the option lines of index urls and trusted hosts
- Add "--emit-trusted-host/--no-emit-trusted-host" option to fix #382

### CODE REVIEW

File: piptools/scripts/compile.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ class PipCommand(pip.basecommand.Command):
               help="Add header to generated file")
 @click.option('--index/--no-index', is_flag=True, default=True,
               help="Add index URL to generated file")
+@click.option('--emit-trusted-host/--no-trusted-host', is_flag=True, default=True,
```
Comment (davidovich) [inline piptools/scripts/compile.py]: To avoid confusion, the --no flag should provide symmetry to the positive version: --no-emit-trusted-host.

Comment (tonyseek) [inline piptools/scripts/compile.py]: agree and done

**Comment by Groxx** [comment]
Since things have changed a bit, think you could rebase?

I'd love to get this in, inconsistent `--trusted-host` lines are the one thing at the moment that prevents dev and CI environments from generating identical *.txt files, and I love having an automated sanity check to catch people who edit `requirements.txt`.  Not sure how to bump it for attention though (beyond updating the code / leaving a comment).

File: piptools/utils.py [hunk 2]
```diff
@@ -203,3 +203,14 @@ def lookup_table(values, key=None, keyval=None, unique=False, use_lists=False):
...
+
+def dedup(iterable):
+    """Deduplicate an iterable object like iter(set(iterable)) but
+    order-reserved.
+    """
+    emitted = set()
+    for x in iterable:
+        if x not in emitted:
+            yield x
+            emitted.add(x)
```
Comment (davidovich) [inline piptools/utils.py]: The merged PR #468, which does a similar fix for index-urls unicity, uses the OrderedDict idiom to get unique entries, could you reuse that?

**Comment by tonyseek** [comment]
@Groxx @davidovich Thanks for your review. The branch is up-to-date now and the revision is pushed.

**Comment by bepetersn** [comment]
Wondering if removing --extra-index-url from requirements.txt output is possible as well.

**Pull request merged**